File: core/src/main/java/com/projecthub/core/enums/TaskStatus.java
package com.projecthub.core.enums;

public enum TaskStatus {
    PENDING,
    IN_PROGRESS,
    COMPLETED,
    CANCELLED,
    FAILED;

    public static boolean isValidStatus(String status) {
        for (TaskStatus ts : TaskStatus.values()) {
            if (ts.name().equals(status)) {
                return true;
            }
        }
        return false;
    }
}

File: src/main/kotlin/com/projecthub/domain/project/TaskStatus.kt
package com.projecthub.domain.project

import com.projecthub.domain.ValueObject
import jakarta.persistence.Embeddable

/**
 * Value object representing the status of a task.
 */
@Embeddable
enum class TaskStatus : ValueObject {
    TODO,
    IN_PROGRESS,
    BLOCKED,
    REVIEW,
    DONE;

    override fun equals(other: Any?): Boolean {
        if (this === other) return true
        if (other !is TaskStatus) return false
        return this.name == other.name
    }

    override fun hashCode(): Int = name.hashCode()

    override fun toString(): String = name
}
